python/observation-publisher/publisher/observation_log_consumer.py

```diff
@@ -0,0 +1,149 @@
+import abc
+from typing import List, Optional, Union, Dict
+
+import numpy as np
+import pandas as pd
+from caraml.upi.v1.prediction_log_pb2 import PredictionLog
+from confluent_kafka import Consumer, KafkaException
+from google.protobuf.internal.well_known_types import ListValue
+
+from publisher.config import (
+    KafkaConsumerConfig,
+    ModelSpec,
+    ValueType,
+    ObservationSourceConfig,
+    ObservationSource,
+)
+from publisher.observability_backend import ObservationSink
+
+
+class PredictionLogConsumer(abc.ABC):
+    @abc.abstractmethod
+    def poll_new_logs(self) -> List[PredictionLog]:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def commit(self):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def close(self):
+        raise NotImplementedError
+
+    def start_polling(self, observation_sink: ObservationSink, model_spec: ModelSpec):
+        try:
+            while True:
+                logs = self.poll_new_logs()
```

Comment:
how is the batching with certain bytes size? is it possible from kafka consumer?

Reply:
No, that's not possible, unfortunately. We will have to implement it ourselves if its necessary.

Reply:
do you plan to update in this PR or separate PR?

Reply:
We could control the number of rows in a batch, so that should have suffice. If it's not, i will implement this in a separate PR.